src/coreclr/pal/inc/pal.h

```diff
@@ -4342,8 +4339,9 @@ PALIMPORT PAL_FILE * __cdecl _wfopen(const WCHAR *, const WCHAR *);
 PALIMPORT int __cdecl rand(void);
 PALIMPORT void __cdecl srand(unsigned int);
 
-PALIMPORT DLLEXPORT int __cdecl printf(const char *, ...);
-PALIMPORT int __cdecl vprintf(const char *, va_list);
+// Forward declare functions that are in header files we can't include yet
```

Comment:
Can you please move these under the `#ifndef PAL_STDCPP_COMPAT` few hundred lines above and then in the `palinternal.h` use the `#define printf DUMMY_printf` / `#undef DUMMY_printf` trick (for both functions)? This ensures that our declaration of these doesn't collide with the ones in the system headers. We hit a problem like that in many cases in the past when on some os-es, some functions have nothrow spec and on some they don't.